File: files/200-299/269/CH-269.R
library(tidyverse)
library(readxl)

path = "files/200-299/269/CH-269 Text Cleaning.xlsx"
input = read_excel(path, range = "B2:B9")
test  = read_excel(path, range = "D2:D9")

result = input %>%
  mutate(ID = str_replace_all(str_remove_all(ID, "-"), "(.)(?=.)", "\\1-"))

all.equal(result$ID, test$`ID 1`, check.attributes = FALSE)
# > [1] TRUE

File: files/200-299/269/CH-269.py
import pandas as pd

path = "200-299/269/CH-269 Text Cleaning.xlsx"
input = pd.read_excel(path, usecols="B", skiprows=1, nrows=8)
test = pd.read_excel(path, usecols="D", skiprows=1, nrows=8)

input["ID 1"] = ["-".join(c for c in name if c != "-") for name in input["ID"]]

print(input["ID 1"].equals(test["ID 1"])) # True